### PR TITLE
feat(form-data): allow passing a form to FormData 

### PR DESCRIPTION
**Pull request by mcansh**

### CODE REVIEW

File: packages/form-data/src/form-data.js [hunk 1]
```diff
@@ -312,3 +318,21 @@ const BlobFile = class File {
 const panic = (error) => {
   throw error;
 };
+
+/**
+ *
+ * @param {Element} element
+ * @returns {element is HTMLSelectElement}
+ */
+function isSelectElement(element) {
```
Comment (ryanflorence) [inline packages/form-data/src/form-data.js:327]: these are only used once, might as well inline them?

Comment (mcansh) [reply packages/form-data/src/form-data.js:327]: doing so loses out on some type checking as it doesn't do a `element is HTMLSelectElement` assertion anymore `Property 'value' does not exist on type 'Element'.` :/

Comment (ryanflorence) [reply packages/form-data/src/form-data.js:327]: TYPESCRIPT!!!!!!

Comment (mcansh) [reply packages/form-data/src/form-data.js:327]: i felt a jsdoc to typescript migration was a little 🤏 outside of the scope of this PR haha

File: packages/form-data/test/form-data.spec.js [hunk 2]
```diff
@@ -259,4 +259,28 @@ export const test = (test) => {
       "rename.md"
     );
   });
+
+  test("Should allow passing a form element",  () => {
```
Comment (ryanflorence) [inline packages/form-data/test/form-data.spec.js]: Let's drop the jsdom dev dependency and just make our own mock `HTMLFormElement` that's got all the right methods/getters/etc. that we use.

File: packages/form-data/test/form-data.spec.js [hunk 3]
```diff
@@ -7,7 +7,7 @@ import { assert } from "./test.js";
  * @param {import('./test').Test} test
  */
 export const test = (test) => {
-  test("test baisc", async () => {
```
Comment (ryanflorence) [inline packages/form-data/test/form-data.spec.js:10]: BAISC!

File: packages/form-data/test/form-data.spec.js [hunk 4]
```diff
@@ -259,4 +259,37 @@ export const test = (test) => {
...
+          form: "my-form",
+        },
+        {
+          tagName: "INPUT",
+          name: "remember-me",
+          value: "on",
+          checked: true,
+        }
+      ]
+    };
```
Comment (ryanflorence) [inline packages/form-data/test/form-data.spec.js]: We need `Object.prototype.toString.call(value) === "[object HTMLFormElement]";` to evaluate correctly.



```js
class FakeForm {
  get [Symbol.toStringTag]() {
    return "HTMLFormElement";
  }
  
  get elements() {
    return [/* ... */]
  }

  get id() {
    return "my-form"
  }
}

let form = new FakeForm();
```

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Symbol/toStringTag

**Pull request merged**